### PR TITLE
[DT] Position fix and new filter to TPs from Phase2 L1T emulator

### PR DESCRIPTION
**Pull request by jfernan2**
#### PR description:

This PR fixes a bug in the positions retrieved by the DT L1T Phase2 emulator (credits to @javi-llorente ) and applies a coincidence filter to the resulting Trigger Primitives  (credits to @dermotmoran ).

It constitutes the v2.1 of the AM algorithm, fixing v2.0 and recovering efficiency losses to the levels of the original v1.0

#### PR validation:

Talks explaining the problem and supporting this change:

Report from J. F de Troconiz: AM v2 Emulator Global Phi / Phi_B Assignment
https://indico.cern.ch/event/1416436/contributions/5953406/attachments/2857442/4999143/DT_TS_Ph2_24_may.pdf

Coincidence filter by D. Moran:
https://indico.cern.ch/event/1425784/contributions/5998104/attachments/2876917/5038456/dt_co.pdf

Talk by J. Llorente explaining the fix in the DT Upgrade meeting:
https://indico.cern.ch/event/1435604/contributions/6040623/attachments/2894593/5074862/llorenteDT.pdf

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@cms-sw/l1-l2  the PR is finally ready for review, thanks for your patience

File: L1Trigger/DTTriggerPhase2/python/dtTriggerPhase2PrimitiveDigis_cfi.py [hunk 1]
```diff
@@ -40,9 +42,8 @@
                                                lut_2sl = cms.FileInPath('L1Trigger/DTTriggerPhase2/data/fitterlut_2sl.dat'),
                                                shift_filename = cms.FileInPath('L1Trigger/DTTriggerPhase2/data/wire_rawId_x.txt'),
                                                shift_theta_filename = cms.FileInPath('L1Trigger/DTTriggerPhase2/data/theta_shift.txt'),
-                                               maxdrift_filename = cms.FileInPath('L1Trigger/DTTriggerPhase2/data/drift_time_per_chamber.txt'),
+                                               maxdrift_filename = cms.FileInPath('L1Trigger/DTTriggerPhase2/data/simple_vdrift.txt'),
```
Comment (aloeliger) [inline L1Trigger/DTTriggerPhase2/python/dtTriggerPhase2PrimitiveDigis_cfi.py:45]: Are these files already committed or in a PR somewhere?

Comment (jfernan2) [reply L1Trigger/DTTriggerPhase2/python/dtTriggerPhase2PrimitiveDigis_cfi.py:45]: Yes, of course, in the corresponding:
https://github.com/cms-data/L1Trigger-DTTriggerPhase2

**Comment by aloeliger** [comment]
+l1

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by aloeliger** [comment]
@cms-sw/upgrade-l2 just a ping on the signature

**Comment by jfernan2** [comment]
@srimanob can you sign this pr please?

**Comment by srimanob** [comment]
+Upgrade

**Comment by srimanob** [comment]
@cmsbuild please test

Just to refresh the test.

**Comment by srimanob** [comment]
Hi @aloeliger @jfernan2 
I think the change is expected, we see change in Phase-2 workflows. One strange is that we see more failure comparisons from DDD on L1T, than DD4hep. I assume it should be the same order. Not so clear why.

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**